tools/etcd-dump-logs/etcd-dump-log_test.go

```diff
@@ -0,0 +1,277 @@
+
+	irralarm := &etcdserverpb.AlarmRequest{3, 4, 5}
+
+	irrauthenable := &etcdserverpb.AuthEnableRequest{}
+
+	irrauthdisable := &etcdserverpb.AuthDisableRequest{}
+
+	irrauthenticate := &etcdserverpb.InternalAuthenticateRequest{"myname", "password", "token"}
+
+	irrauthuseradd := &etcdserverpb.AuthUserAddRequest{"name1", "pass1"}
+
+	irrauthuserdelete := &etcdserverpb.AuthUserDeleteRequest{"name1"}
+
+	irrauthuserget := &etcdserverpb.AuthUserGetRequest{"name1"}
+
+	irrauthuserchangepassword := &etcdserverpb.AuthUserChangePasswordRequest{"name1", "pass2"}
+
+	irrauthusergrantrole := &etcdserverpb.AuthUserGrantRoleRequest{"user1", "role1"}
+
+	irrauthuserrevokerole := &etcdserverpb.AuthUserRevokeRoleRequest{"user2", "role2"}
+
+	irrauthuserlist := &etcdserverpb.AuthUserListRequest{}
+
+	irrauthrolelist := &etcdserverpb.AuthRoleListRequest{}
+
+	irrauthroleadd := &etcdserverpb.AuthRoleAddRequest{"role2"}
+
+	irrauthroledelete := &etcdserverpb.AuthRoleDeleteRequest{"role1"}
+
+	irrauthroleget := &etcdserverpb.AuthRoleGetRequest{"role3"}
+
+	perm := &authpb.Permission{
+		PermType: authpb.WRITE,
+		Key:      []byte("Keys"),
+		RangeEnd: []byte("RangeEnd"),
+	}
+
+	irrauthrolegrantpermission := &etcdserverpb.AuthRoleGrantPermissionRequest{"role3", perm}
+
+	irrauthrolerevokepermission := &etcdserverpb.AuthRoleRevokePermissionRequest{"role3", []byte("key"), []byte("rangeend")}
+
+	irrs := []etcdserverpb.InternalRaftRequest{
+		{ID: 5, Range: irrrange},
+		{ID: 6, Put: irrput},
+		{ID: 7, DeleteRange: irrdeleterange},
+		{ID: 8, Txn: irrtxn},
+		{ID: 9, Compaction: irrcompaction},
+		{ID: 10, LeaseGrant: irrleasegrant},
+		{ID: 11, LeaseRevoke: irrleaserevoke},
+		{ID: 12, Alarm: irralarm},
+		{ID: 13, AuthEnable: irrauthenable},
+		{ID: 14, AuthDisable: irrauthdisable},
+		{ID: 15, Authenticate: irrauthenticate},
+		{ID: 16, AuthUserAdd: irrauthuseradd},
+		{ID: 17, AuthUserDelete: irrauthuserdelete},
+		{ID: 18, AuthUserGet: irrauthuserget},
+		{ID: 19, AuthUserChangePassword: irrauthuserchangepassword},
+		{ID: 20, AuthUserGrantRole: irrauthusergrantrole},
+		{ID: 21, AuthUserRevokeRole: irrauthuserrevokerole},
+		{ID: 22, AuthUserList: irrauthuserlist},
+		{ID: 23, AuthRoleList: irrauthrolelist},
+		{ID: 24, AuthRoleAdd: irrauthroleadd},
+		{ID: 25, AuthRoleDelete: irrauthroledelete},
+		{ID: 26, AuthRoleGet: irrauthroleget},
+		{ID: 27, AuthRoleGrantPermission: irrauthrolegrantpermission},
+		{ID: 28, AuthRoleRevokePermission: irrauthrolerevokepermission},
+	}
+
+	for i, irr := range irrs {
+		var currentry raftpb.Entry
+		currentry.Term = uint64(i + 4)
+		currentry.Index = uint64(i + 10)
+		currentry.Type = raftpb.EntryNormal
+		currentry.Data = pbutil.MustMarshal(&irr)
+		*ents = append(*ents, currentry)
+	}
+}
+
+func appendUnknownNormalEnts(ents *[]raftpb.Entry) {
+	var currentry raftpb.Entry
+	currentry.Term = 27
+	currentry.Index = 34
+	currentry.Type = raftpb.EntryNormal
+	currentry.Data = []byte("?")
+	*ents = append(*ents, currentry)
+}
```

Comment:
I added the check to make sure unknown normal entries are listed as well, like the behavior of the tool before my change.